frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+		// TODO: publicAddresses is still not loaded. 
+		// Solve temporarily by waiting for 0.5 sec.
+		await delayMsec(500);
+		const provider = await detectEthereumProvider();
+		let details = [];
+		if (!provider) {
+			setErrorNoMetamaskMessage("Couldn't get data from Ethereum Provider. Please install MetaMask!");
+			console.error("no ethereum provider");
+			// Display only ETH address from backend
+			publicAddresses.map(pubAddress => {
+				details.push({ "pubAddress": pubAddress, "balance": NaN });
+			}) ;
+			setDetailData(details);
+			return;
+		}
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				details.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(details);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Account (Public Address)
+						</Typography>
+					</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Balance (Eth)
+						</Typography>
+					</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value} />;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+/**
+ * Component to display and add Eth address from Wallet. Currently only Metamask is supported.
+ * @param user user object of the currently logged in user/admin
+ * @returns {JSX.Element}
+ */
+export default function Wallet({ user }) {
+	const [loadedAddresses, setLoadedAddresses] = useState(false);
+	const [publicAddresses, setPublicAddresses] = useState([]);
+	const [errorNoMetamaskMessage, setErrorNoMetamaskMessage] = useState("");
+
+	const history = useHistory();
+
+	const routeChange = (path) =>{
+		history.push(path);
+	};
+
+	const getEthAddressesFromServer = () => {
+		if (!loadedAddresses) {
+			appwriteApi.getOwnEthAddress(user.$id)
+				.then(result => {
+					// Set the address(es)
+					let currentPubAddr = publicAddresses;
+					for (let i = 0; i < result.sum; i++) {
+						currentPubAddr.push(result.documents[i].walletAddress);
+					}
+					setPublicAddresses(currentPubAddr);
+					setLoadedAddresses(true);
+				})
+				.catch((error) => {
+					console.error(error);
+				});
+		}
+	};
+
+	useEffect(() => {
+		getEthAddressesFromServer();
+	}, []);
+
+	if (!user){
+		routeChange("/");
+		return <></>;
+	}
+
+	/* Handle button connect to MetaMask */
```

Comment:
This comment actually is a function name at least by commonly taught software engineering.
Same for line 158.
Comments are not bad but their purpose is different from this. If you'd like to explain what your code snippet does (which is very legitimate) then function names are a better choice.

Single-statement functions are perfectly fine, when there is a need to comment that statement.